### PR TITLE
gdal: 3.5.2 -> 3.6.0.1

### PR DESCRIPTION
**Pull request by dotlambda**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

closes https://github.com/NixOS/nixpkgs/pull/201834

### CODE REVIEW

File: pkgs/development/libraries/lerc/default.nix [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
     repo = "lerc";
     rev = "v${version}";
-    hash = "sha256-QO5+ouQy5nOcAgvxMeBDoSBP+G3ClDjXipnkuSIDcP0=";
+    hash = "sha256-IHY9QtNYsxPz/ksxRMZGHleT+/bawfTYNVRSTAuYQ7Y=";
   };
 
+  patches = [
+    # https://github.com/Esri/lerc/pull/227
+    (fetchpatch {
+      url = "https://github.com/Esri/lerc/commit/5462ca7f7dfb38c65e16f5abfd96873af177a0f8.patch";
```
Comment (risicle) [inline pkgs/development/libraries/lerc/default.nix:24]:
```suggestion
      name = "use-cmake-install-full.patch";
      url = "https://github.com/Esri/lerc/commit/5462ca7f7dfb38c65e16f5abfd96873af177a0f8.patch";
```

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 201730` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.worldengine</li>
    <li>python39Packages.worldengine</li>
    <li>worldengine-cli</li>
  </ul>
</details>
<details>
  <summary>25 packages failed to build:</summary>
  <ul>
    <li>apacheHttpdPackages.mod_tile (apacheHttpdPackages_2_4.mod_tile)</li>
    <li>mysql-workbench</li>
    <li>paraview</li>
    <li>perl534Packages.Tirex</li>
    <li>perl536Packages.Tirex</li>
    <li>python310Packages.asf-search</li>
    <li>python310Packages.bsuite</li>
    <li>python310Packages.cartopy</li>
    <li>python310Packages.fiona</li>
    <li>python310Packages.geopandas</li>
    <li>python310Packages.osmnx</li>
    <li>python310Packages.plotnine</li>
    <li>python310Packages.rlax</li>
    <li>python310Packages.wktutils</li>
    <li>python39Packages.asf-search</li>
    <li>python39Packages.bsuite</li>
    <li>python39Packages.cartopy</li>
    <li>python39Packages.fiona</li>
    <li>python39Packages.geopandas</li>
    <li>python39Packages.osmnx</li>
    <li>python39Packages.plotnine</li>
    <li>python39Packages.rlax</li>
    <li>python39Packages.wktutils</li>
    <li>qgis-ltr</li>
    <li>t-rex</li>
  </ul>
</details>
<details>
  <summary>34 packages built:</summary>
  <ul>
    <li>cloudcompare</li>
    <li>entwine</li>
    <li>gdal (python310Packages.gdal)</li>
    <li>gmt</li>
    <li>gplates</li>
    <li>grass</li>
    <li>lerc</li>
    <li>libLAS</li>
    <li>mapcache</li>
    <li>mapnik</li>
    <li>mapproxy</li>
    <li>mapserver</li>
    <li>merkaartor</li>
    <li>openorienteering-mapper</li>
    <li>pdal</li>
    <li>postgresql11Packages.postgis</li>
    <li>postgresql12Packages.postgis</li>
    <li>postgresql13Packages.postgis</li>
    <li>postgresqlPackages.postgis (postgresql14Packages.postgis)</li>
    <li>postgresql15Packages.postgis</li>
    <li>python310Packages.pygmt</li>
    <li>python310Packages.python-mapnik</li>
    <li>python310Packages.rasterio</li>
    <li>python39Packages.gdal</li>
    <li>python39Packages.pygmt</li>
    <li>python39Packages.python-mapnik</li>
    <li>python39Packages.rasterio</li>
    <li>pytrainer</li>
    <li>qgis</li>
    <li>qlandkartegt</li>
    <li>qmapshack</li>
    <li>saga</li>
    <li>sumo</li>
    <li>udig</li>
  </ul>
</details>

**Comment by risicle** [comment]
One test fails for `fiona` on aarch64-linux: https://gist.github.com/risicle/2631e23beaf492e95d6d1ef44d3d789e

**Comment by risicle** [comment]
@ofborg build python310Packages.fiona

**Comment by dotlambda** [comment]
> One test fails for `fiona` on aarch64-linux: https://gist.github.com/risicle/2631e23beaf492e95d6d1ef44d3d789e

I was too lazy to investigate and just opened an upstream issue: https://github.com/Toblerity/Fiona/issues/1164

**Comment by dotlambda** [comment]
I actually get the same error on x86_64-linux.

**Comment by risicle** [comment]
Note the ofborg test failures of `test_tiff_read_multi_threaded_vsicurl` on macos. I also get these on macos 10.15, but note upstream appear to skip these on macos too: https://github.com/OSGeo/gdal/blob/ccc05bc95d73217e85117bea84efbfe6a7ac2e70/autotest/gcore/tiff_read.py#L4856

Would be nice if either upstream used a more universal mechanism to detect macos or maybe we just found a way to emulate it via this `gdaltest.is_travis_branch` mechanism.

**Comment by risicle** [comment]
`nixpkgs-review` reveals no new failures on macos 10.15.

**Comment by wegank** [comment]
Tests seem to fail again on aarch64-darwin on Hydra: https://hydra.nixos.org/build/201653821/nixlog/1

**Pull request merged**